### PR TITLE
fix: defer setting default mode to core

### PR DESCRIPTION
**Pull request by anshumanv**
**What kind of change does this PR introduce?**

fix

**Did you add tests for your changes?**
Yes

**If relevant, did you update the documentation?**
Not needed I think

**Summary**
- Defer setting default mode to core so it can print warnings and stuff

**Does this PR introduce a breaking change?**
Nope

**Other information**

### CODE REVIEW

**Review by evilebottnawi — APPROVED** [review]

**Review by snitin315 — DISMISSED** [review]

**Review by rishabh3112 — DISMISSED** [review]

**Review by snitin315 — APPROVED** [review]

**Review by evilebottnawi — APPROVED** [review]
Feel free to merge on green

File: test/colors/colors.test.js [hunk 1]
```diff
@@ -26,6 +26,8 @@ describe('colors related tests', () => {
         const { stderr, stdout, exitCode } = run(__dirname, [
             '--stats=verbose',
             `--config=${resolve(__dirname, './no-stats.webpack.config.js')}`,
+            '--mode',
+            'production',
```
Comment (snitin315) [inline test/colors/colors.test.js:30]: I think it will be better to define `mode: production` inside the config.

Comment (evilebottnawi) [reply test/colors/colors.test.js:30]: Yep, let's do it

Comment (anshumanv) [reply test/colors/colors.test.js:30]: Yep, let's do it

**Comment by evilebottnawi** [comment]
@anshumanv We need fix test for stats and watch, let's add `mode: development`

**Comment by anshumanv** [comment]
should be good now

**Review by evilebottnawi — APPROVED** [review]

**Comment by evilebottnawi** [comment]
Thanks

**Pull request merged**